### PR TITLE
[Hotfix][Transform][Sql] function concat_ws handle array type， avoid write [Ljava.lang.String.

### PR DESCRIPTION
**Pull request by sohurdc**
### Purpose of this pull request

When transform from Hive to MySQL, if there are columns of array type, using the concat_ws function for transformation will result in writing [Ljava.lang.String. This is because the array type does not implement the toString method, so special handling is required. Loop through and process the array type one by one.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

I added unit tests, but did not add e2e tests.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/sql/zeta/ConcatWsFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ConcatWsFunctionTest {
+
+    @Test
+    public void testConcatWs() {
```
Comment (Hisoka-X) [inline seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/sql/zeta/ConcatWsFunctionTest.java:32]: Thanks @sohurdc ! Could you add some null value in array to test?

Comment (sohurdc) [reply seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/sql/zeta/ConcatWsFunctionTest.java:32]: ok

**Review by Hisoka-X — DISMISSED** [review]
LGTM. Thanks @sohurdc !

**Comment by Hisoka-X** [comment]
Please re-try failed ci.

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**